### ISSUE TITLE
undefined method `env' for Chamber:Module - Rails 6

### ISSUE THREAD

**Issue by veverkap** [thekompanee/chamber#72]
## Current Behavior
Trying to use Chamber in a Rails initializer and the Rails application will not boot.

```
	 2: from /Users/patrickveverka/Code/personal/blah/config/application.rb:9:in `<main>'
	 1: from /Users/patrickveverka/Code/personal/blah/config/application.rb:11:in `<module:Blah>'
/Users/patrickveverka/Code/personal/blah/config/application.rb:14:in `<class:Application>': private method `load' called for Chamber:Module (NoMethodError)
```

### Steps to Reproduce
1. Create a new Rails app (rails new blah)
2. Add `gem "chamber"` to Gemfile
3. bundle install
4. Add call to `Chamber.env.setting` in an initializer
5. Run local server `rails s`
6. App go boom

## Expected Behavior
I thought that the chamber secrets would be available in initializers (devise, mailer, etc)

## Environment

 - OS:

    - [ ] Windows
    - [x] macOS
    - [ ] iOS
    - [ ] Android
    - [ ] Linux
    - [ ] Other

 - OS Version: 10.15.6 (Catalina)

I made an example app at https://github.com/veverkap/chamber_error that shows the error at boot.  I included the private keys in the repo but will destroy it later.

**Comment by jfelchner** [comment]
Ugh.  Rails and backwards incompatibility. 😂

I appreciate you taking a look at this and for the reproduction.  I'll try to take a gander at this and get it fixed.

**Comment by jfelchner** [comment]
Also sorry it took so long for me to see this!

**Comment by drselump14** [comment]
Hi guys, is there any progress or workaround with this issue? any help would be appreciated

**Comment by drselump14** [comment]
@veverkap Hi, do you know a workaround to address this issue?

**Comment by veverkap** [comment]
Nope, I ended up not using Chamber.

**Comment by drselump14** [comment]
Thank you for the reply!!  That's unfortunate.. I think this gem is the
best solution for rails configuration

-- 
slamet kristanto

**Comment by drselump14** [comment]
Rails  6.0 using zeitwerk for default, that's probably causing the issue

**Comment by jfelchner** [comment]
Sorry all, I've had some other things came up.  I'll work on it this weekend.

**Comment by drselump14** [comment]
Great! Thanks Jeff

**Comment by jfelchner** [comment]
Worked on this a bit but didn't quite get there, but I'm on it.

**Comment by pcreux** [comment]
Just ran into the same issue. `chamber` requires `thor (>= 0.19.1, < 0.21)` while rails 6 requires `thor (>= 0.20.3, < 2.0)`. So `bundler` happily installed `chamber v0.0.1` since it's does not require a specific version of thor – but it's blank.

So this issue is definitely related to https://github.com/thekompanee/chamber/issues/73.

`chamber` looks awesome but I'll have to find an alternative due to the lack of maintenance.

**Comment by jfelchner** [comment]
@pcreux updating thor is not a simple matter as the thor integration chamber uses was broken in `thor < 0.21`. I'm currently working on it, but we all have day jobs.

**Comment by jfelchner** [comment]
@pcreux additionally if `thor` was the dependency problem, then it should have installed it with `0.20.3` since that matches both `chamber` as well as `rails`.  My guess is that there's something else that's the problem and not the `thor` dependency.

Regardless, I'm still upgrading Chamber to work with later versions of `thor`.

**Comment by pcreux** [comment]
@jfelchner Right, 0.20.3 should have worked... 🤔

> but we all have day jobs

Isn't chamber sponsored by TheKompanee? 🙂

Thank you for your hard work on this project.

**Comment by jfelchner** [comment]
@veverkap your sample app is using `chamber` version `0.0.4`.  The latest version is `2.13.1`.  When you use the latest version, it works fine.

**Comment by jfelchner** [comment]
To be more clear, you and @pcreux had the same issue because you were only trying to update `chamber`.  Instead you should have been updating `chamber` AND `thor`, which would have set the version to `thor` `0.20.3`.

So @pcreux the issue wasn't with Chamber after all, you can take back your "unmaintained" comment now. ;)

@veverkap The diff you can apply to your sample app is as follows:

```diff
diff --git i/Gemfile w/Gemfile
index 06446f0..69ddc7a 100644
--- i/Gemfile
+++ w/Gemfile
@@ -22,11 +22,11 @@ gem 'jbuilder', '~> 2.7'
 # Use Active Model has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
 
 # Use Active Storage variant
 # gem 'image_processing', '~> 1.2'
-gem "chamber"
+gem "chamber", '= 2.13.1'
 # Reduces boot times through caching; required in config/boot.rb
 gem 'bootsnap', '>= 1.4.2', require: false
 
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
diff --git i/Gemfile.lock w/Gemfile.lock
index f0981d9..309df39 100644
--- i/Gemfile.lock
+++ w/Gemfile.lock
@@ -69,20 +69,21 @@ GEM
       nokogiri (~> 1.8)
       rack (>= 1.6.0)
       rack-test (>= 0.6.3)
       regexp_parser (~> 1.5)
       xpath (~> 3.2)
-    chamber (0.0.4)
-      hashie (~> 2.0)
+    chamber (2.13.1)
+      hashie (~> 3.3)
+      thor (>= 0.19.1, < 0.21)
     childprocess (3.0.0)
     concurrent-ruby (1.1.7)
     crass (1.0.6)
     erubi (1.9.0)
     ffi (1.13.1)
     globalid (0.4.2)
       activesupport (>= 4.2.0)
-    hashie (2.1.2)
+    hashie (3.6.0)
     i18n (1.8.5)
       concurrent-ruby (~> 1.0)
     jbuilder (2.10.0)
       activesupport (>= 5.0.0)
     listen (3.2.1)
@@ -167,11 +168,11 @@ GEM
     sprockets-rails (3.2.1)
       actionpack (>= 4.0)
       activesupport (>= 4.0)
       sprockets (>= 3.0.0)
     sqlite3 (1.4.2)
-    thor (1.0.1)
+    thor (0.20.3)
     thread_safe (0.3.6)
     tilt (2.0.10)
     turbolinks (5.2.1)
       turbolinks-source (~> 5.2)
     turbolinks-source (5.2.0)
@@ -202,11 +203,11 @@ PLATFORMS
 
 DEPENDENCIES
   bootsnap (>= 1.4.2)
   byebug
   capybara (>= 2.15)
-  chamber
+  chamber (= 2.13.1)
   jbuilder (~> 2.7)
   listen (~> 3.2)
   puma (~> 4.1)
   rails (~> 6.0.3, >= 6.0.3.2)
   sass-rails (>= 6)
```

**Comment by jfelchner** [comment]
The next version of `chamber` will be compatible with `thor` `0.19.x` through `2.0`.

**Comment by jfelchner** [comment]
`chamber` `2.14.0` has been released: https://rubygems.org/gems/chamber/versions/2.14.0

**Comment by pcreux** [comment]
@jfelchner That's fantastic!

I didn't want to offence you with my comment - just wanted to share the reason why I wouldn't use this gem if that was of any importance for the company sponsoring the project.

**Comment by jfelchner** [comment]
hahaha no worries.  I'm glad you find it useful. :)

**Comment by jfelchner** [comment]
Additionally I just release 2.14.2 which has all the deprecation warnings for 3.0.0.  I also just released 3.0.0-rc1.  You can get upgrade notes here: https://github.com/thekompanee/chamber/wiki/Upgrading-To-Chamber-3.0
